modules/patterns/pages/centralize-pipeline-definitions.adoc

```diff
@@ -0,0 +1,146 @@
+= Pattern: Centralizing Tekton Pipeline Definitions in Your Repository
+
+This pattern describes how to extract a common Tekton `Pipeline` definition into a local file within your application's Git repository (e.g., `.tekton/build-pipeline.yaml`). Your `PipelineRun` files can then reference this local pipeline, promoting consistency, simplifying maintenance, and enabling version control of the pipeline logic alongside your application code.
+
+This approach is particularly beneficial when you need to customize a standard pipeline or ensure that all build triggers (e.g., push, pull request) use the exact same underlying pipeline tasks and structure.
+
+.Use Cases
+* Customizing a standard {ProductName} build pipeline (e.g., `docker-build-multi-platform-oci-ta`) with additional tasks or modified parameters.
+* Ensuring all `PipelineRun` files (for pushes, pull requests, manual triggers) use an identical, version-controlled pipeline definition.
+* Simplifying the maintenance of pipeline logic by having a single source of truth within the repository.
+* Facilitating easier implementation of other patterns, such as <<Pattern: Achieving Label and Tag Parity for Versioning>>, by providing a single pipeline file to modify.
+
+.Prerequisites
+* Your application has been onboarded to {ProductName}.
+* {ProductName} (via Pipelines-as-Code) has generated initial `PipelineRun` files in your repository's `.tekton/` directory, or you have `PipelineRun` files that reference a remote Tekton Hub or bundle.
+* You have identified the base Tekton `Pipeline` specification you intend to use and customize. This might be from:
+    * The `spec.pipelineSpec` embedded in one of your existing `PipelineRun` files.
+    * A standard pipeline definition from the {ProductName} Tekton catalog (e.g., `docker-build-multi-platform-oci-ta`).
+
+.Procedure
+
+. **Create a Local Pipeline Definition File:**
+   In your repository, create a new YAML file within the `.tekton/` directory, for example, `.tekton/build-pipeline.yaml`. This file will contain your `kind: Pipeline` resource.
+
+. **Define the `Pipeline` Resource:**
+   Populate `.tekton/build-pipeline.yaml` with the `Pipeline` definition.
```

Comment:
Can we give the reader a way to do this with `yq` for less copy/paste burden?

Something like:

```
yq .spec.pipelineSpec old-component-push.yaml | sed 's/^/  /' >> build-pipeline.yaml
```

Reply:
Thanks for the suggestion.

Reply:
I added it to the bottom of this page.